patches/linux/x86_64/0001-x86-boot-clean-up-headers.patch

```diff
@@ -0,0 +1,50 @@
+From 6cf81b4f510dc6e75024065a3bc6d5805d304827 Mon Sep 17 00:00:00 2001
+From: Nick Desaulniers <ndesaulniers@google.com>
+Date: Fri, 1 Mar 2019 15:29:40 -0800
+Subject: [PATCH] x86/boot: clean up headers
+
+The inclusion of <linux/kernel.h> was causing issue as the definition of
+__arch_hweight64 from arch/x86/include/asm/arch_hweight.h eventually gets
+included. The definition is problematic when compiled with -m16 (all code
+in arch/x86/boot/ is) as the "D" inline assembly constraint is rejected
+by both compilers when passed an argument of type long long (regardless
+of signedness, anything smaller is fine).
+
+Because GCC performs inlining before semantic analysis, and
+__arch_hweight64 is dead in this translation unit, GCC does not report
+any issues at compile time.  Clang does the semantic analysis in the
+front end, before inlining (run in the middle) can determine the code is
+dead. I consider this another case of PR33587, which I think we can do
+more work to solve.
+
+It turns out that arch/x86/boot/string.c doesn't actually need
+linux/kernel.h, simply linux/limits.h and linux/types.h. Include them,
```

Comment:
should be `simply linux/limits.h and linux/compiler.h`.  `linux/types.h` was already included and was simply sorted, not added.